src/main.js

```diff
@@ -62,18 +65,18 @@ function isHidden(e) {
 }
 
 function getTextFromElement(e) {
-  return e.innerText === "" ? e.dataset.content : e.innerText;
+  return (e.innerText === "" ? e.dataset.content : e.innerText) || "";
 }
 
 function getVisibleText(e) {
   if (isHidden(e)) {
     // stop if this is hidden
-    return [];
+    return "";
   }
   const children = e.querySelectorAll(":scope > *");
   if (children.length !== 0) {
     // more level => recursive
-    return Array.prototype.slice.call(children).map(getVisibleText).flat();
+    return (e.dataset.content || "") + Array.prototype.slice.call(children).map(getVisibleText).flat().join("");
```

Comment:
In addition to changing the result to be always a string, I've included the `data-content` of the element we investigate because in the new layout they have a `span` with the "S" and the rest scrambled spans are nested inside. The resustling Visible Text before this change would be "ponsored", without the "S".

Reply:
I included `data-content` in `getTextFromElement()`. Can we rewrite the logic of `getTextFromElement()` to make it work?

Reply:
using `getTextFromElement(e)` instead of `(e.dataset.content || "")` worked with the same results for me but the `getVisibleText(e)` result in case the element was actually a timestamp was like "40 mins40 mins40 mins40 mins" because of the `innerText` with nested elements. For "Sponsored" posts, the result was correct.

Reply:
The only refactor that crossed my mind was to add an extra argument to exclude the innerText from the result but I think it was an overkill.

Reply:
Maybe you need to write a new get text for FB5?

Reply:
So the FB5 does not hide character in `<span/>` but it is in `data-content`?

Reply:
Have a look of what the new "sponsored" dom looks like. On my tests, it has happened that the highlighted `data-content` of the outer span was "S". the change is not affecting the classic design implementation.
![image](https://user-images.githubusercontent.com/1640100/78813338-29841500-79cd-11ea-8570-a99f3d8e5a1f.png)
You can switch between the Classic and new Facebook design from the Settings drop down menu on the top right corner of Facebook. You should see a "Switch to New Facebook" option just above "Log Out"

![image](https://user-images.githubusercontent.com/1640100/78813599-8e3f6f80-79cd-11ea-8675-4f9a25613d5d.png)

Reply:
Yeah I don't have that option yet. I tried today with two different machines and two different accounts.

Reply:
I understand it better now with dom picture. Thanks.

Reply:
Could we add a function that get text from parent, and then we concat that with a text from children?

We can reuse `getTextFromElement` to get text from parent or just a function that return `e.dataset.content` for now. This way it will be clearer that,
```
    return (e.dataset.content || "") + Array.prototype.slice.call(children).map(getVisibleText).flat().join("");
```
is basically
```
    return textFromContainerNode() + Array.prototype.slice.call(children).map(getVisibleText).flat().join("");
```

Reply:
Would that work?

Reply:
what's why i needed to add the `(e.dataset.content || "") + ` to include the `data-content` of the inspecting node with child elements and since i needed to concatenate in a single line with an array, I moved the `.join("")` part inside the `getVisibleText` to always return a string

Reply:
yes it would work just fine.